### PR TITLE
(#893) Test for comparison of MapEnvelope and derived class

### PR DESCRIPTION
**Pull request by scratchboom**
Comparison of MapEnvelope with derived classes already supported. I've added the test for that.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1056](https://github.com/yegor256/cactoos/issues/1056) is now in scope, role is `REV`

File: src/test/java/org/cactoos/map/MapEnvelopeTest.java [hunk 1]
```diff
@@ -313,4 +314,17 @@ public void hashCodeFailsOnNull() {
             new MapEntry<>("key11", null);
         new MapOf<String, String>(first, second).hashCode();
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void mapEnvelopeShouldCompareDerivedClasses() {
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @scratchboom Please, test if an empty `MapEnvelope` and an empty derived are equals.

Comment (scratchboom) [reply src/test/java/org/cactoos/map/MapEnvelopeTest.java]: @fabriciofx see new changes

**Comment by llorllale** [comment]
@fabriciofx ping

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/1056#issuecomment-468921423))

@llorllale The user @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#1056](https://github.com/yegor256/cactoos/issues/1056), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#1056](https://github.com/yegor256/cactoos/issues/1056) is assigned to @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr), here is [why](https://www.0crat.com/footprint/C63314D6Z/eb809112-3c66-4f33-9fed-3b791d7e92b1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#1056` already assigned to @iprogrammerr, can't assign to @iprogrammerr

**Comment by Iprogrammerr** [comment]
@scratchboom ,@llorllale Good to merge.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31):

**Comment by 0crat** [comment]
The job [#1056](https://github.com/yegor256/cactoos/issues/1056) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Pull request merged**